projects/bp-gallery/src/components/views/location-curating/LocationManagementDialog.tsx

```diff
@@ -0,0 +1,392 @@
+                      }}
+                      createParentMutation={newLocationTagMutation}
+                    />
+                  </PictureInfoField>
+                </div>
+              </div>
+            </div>
+            <div className='location-management-right'>
+              <div className='location-management-map'></div>
+              <div className='location-management-actions'>
+                <div className='location-management-picture-count'>
+                  {flattenedPictures &&
+                    t('tag-panel.location-pictures', {
+                      count: flattenedPictures.locationTag.pictures.length,
+                    })}
+                </div>
+                <Button
+                  className='location-management-show-pictures-button'
+                  onClick={() => {
+                    handleClose(undefined);
+                    history.push(`/show-more/location/${locationTag.id}`, {
+                      showBack: true,
+                    });
+                  }}
+                  endIcon={<ArrowForwardIos />}
+                >
+                  {t('common.show-pictures')}
+                </Button>
+                {localVisibility ? (
+                  <Button
+                    className='location-management-show-button'
+                    onClick={() => {
+                      setVisible(false);
+                      setLocalVisibility(false);
+                    }}
+                    endIcon={<Visibility />}
+                  >
+                    {t('common.visible')}
+                  </Button>
+                ) : (
+                  <Button
+                    className='location-management-not-show-button'
+                    onClick={() => {
+                      setVisible(true);
+                      setLocalVisibility(true);
+                    }}
+                    endIcon={<VisibilityOff />}
+                  >
+                    {t('common.invisible')}
+                  </Button>
+                )}
+                {isRoot ? (
+                  <Button
+                    className='location-management-root-button'
+                    onClick={() => {
+                      if (locationTag.parent_tags?.length) {
+                        setTagAsRoot(false);
+                        setIsRoot(false);
+                      }
+                    }}
+                    endIcon={<AccountTree />}
+                  >
+                    {t('common.root')}
+                  </Button>
+                ) : (
+                  <Button
+                    className='location-management-not-root-button'
+                    onClick={() => {
+                      setTagAsRoot(true);
+                      setIsRoot(true);
+                    }}
+                    endIcon={<AccountTree />}
+                  >
+                    {t('common.no-root')}
+                  </Button>
+                )}
```

Comment:
Maybe compress this code by using ternaries inside the button content, icon and classname? My suggestion in the scss might help